plugins/modules/net_tools/ldap/ldap_attrs.py

```diff
@@ -263,9 +264,10 @@ def exact(self):
     def _is_value_present(self, name, value):
         """ True if the target attribute has the given value. """
         try:
-            is_present = bool(
-                self.connection.compare_s(self.dn, name, value))
-        except ldap.NO_SUCH_ATTRIBUTE:
+            filterstr = "(%s=%s)" % (name, value.decode())
+            dns = self.connection.search_s(self.dn, ldap.SCOPE_BASE, filterstr)
+            is_present = bool(len(dns) == 1)
```

Comment:
The `bool` is redundant in this statement, as the result of the `==` condition is already a `bool`. If you want to keep the `(...)` around it do clarify, fine:
```suggestion
            is_present = (len(dns) == 1)
```

Reply:
I know it is redundant, I just mimicked the original `bool(self.connection.compare_s())` that also bools a returnvalue that is already a bool?

Reply:
I had not looked much into the unchanged code in the module. Yes, it seems the the `compare_s()` function does return a `bool` [1], which makes the  `bool()` around it quite silly, IMO. If I may suggest, remove it from there as well.

[1] https://www.python-ldap.org/en/python-ldap-3.4.3/reference/ldap.html#ldap.LDAPObject.compare_s

Reply:
Maybe some older version did not; I would not touch that line unless someone really verified that *every single version* of the Python LDAP bindings actually returns a boolean.

Reply:
Please pay attention here, that line is _replaced_ by my code...